### PR TITLE
PIMS-2816 Run sonar scan from github actions

### PR DESCRIPTION
**Pull request by asanchezr**
This PR adds the ability to run SonarQube scans for pull requests and submit the results to the PIMS dashboard at https://sonarqube-jcxjin-tools.pathfinder.gov.bc.ca

It also adds a comment to the PR with a summary of the scan (badges)

### CODE REVIEW

File: backend/api/Pims.Api.csproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <AssemblyVersion>0.1.0.0</AssemblyVersion>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <NoWarn>$(NoWarn);1591</NoWarn>
+    <ProjectGuid>FA2CFDF0-DB90-4D54-813D-8BAD039A74F6</ProjectGuid>
```
Comment (asanchezr) [inline backend/api/Pims.Api.csproj:11]: SonarQube for NET Core cannot scan projects unless they have valid GUIDs. NET Core doesn't use them but we need to add it to please SonarQube...

File: openshift/scripts/common.sh [hunk 2]
```diff
@@ -37,10 +37,12 @@ source "$(dirname ${0})/envars"
...
-$(oc whoami &>/dev/null) || {
-  echo -e "\nPlease verify oc login\n"
-  exit 1
-}
+if [ -z "${SKIP_OC_LOGIN:-}" ]; then
+  $(oc whoami &>/dev/null) || {
+    echo -e "\nPlease verify oc login\n"
+    exit 1
+  }
+fi
```
Comment (asanchezr) [inline openshift/scripts/common.sh:45]: Provide a way to skip logging into Openshift as this is not yet supported from github actions

File: openshift/scripts/oc-sonar-scan.sh [hunk 3]
```diff
@@ -51,8 +51,8 @@ CMD_TEST="npm run coverage"
 CMD_SONAR_SCAN="sonar-scanner \
   -Dsonar.host.url='${SONAR_URL}' \
   ${SONAR_TOKEN:+ -Dsonar.login='${SONAR_TOKEN}'} \
-  -Dsonar.projectKey='pims-frontend-${TARGET}' \
-  -Dsonar.projectName='PIMS Frontend [${TARGET}]'"
+  -Dsonar.projectKey='${SONAR_PROJECT_KEY}' \
+  -Dsonar.projectName='${SONAR_PROJECT_NAME}'"
```
Comment (asanchezr) [inline openshift/scripts/oc-sonar-scan.sh:55]: Consistent naming with backend sonar project

**Comment by asanchezr** [comment]
For some reason the actions are failing. I will have to debug this on Monday. Do not merge please

File: backend/api/Pims.Api.csproj (hunk 1 above)
Comment (Fosol) [reply backend/api/Pims.Api.csproj:11]: Just a warning about these.  If for some reason our branches get out of sync (different devs using different guids), it causes havoc when using Visual Studio.

File: backend/tests/unit/api/coverlet.runsettings [hunk 4]
```diff
@@ -6,7 +6,7 @@
         <Configuration>
           <Format>json,opencover</Format>
           <Exclude>[*.Test]*,[*]*Model,[*]*Migrations*</Exclude> <!-- [Assembly-Filter]Type-Filter -->
-          <ExcludeByAttribute>Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute</ExcludeByAttribute>
+          <ExcludeByAttribute>Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute,ExcludeFromCodeCoverage</ExcludeByAttribute>
```
Comment (Fosol) [inline backend/tests/unit/api/coverlet.runsettings:9]: Thanks!

**Comment by asanchezr** [comment]
Closing this PR. 

At this moment there are restrictions on PRs coming from forked repositories - GitHub will not set the secrets on those PRs and as such the actions that were configured here wouldn't be able to run as they need the `SONAR_TOKEN` and another token to be able to post a comment on the PR.

**Pull request closed without merging**